src/BinarySerializationTestForStructs.zig

```diff
@@ -0,0 +1,135 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const file = std.fs.File;
+const cwd = std.fs.cwd();
+const fba = std.heap.FixedBufferAllocator;
+
+const ArbitraryType = struct {
+    size: u64,
+    line: []u8
+};
+const BoxedStructure = packed struct {
+    flag: bool,
+    letter: u8,
+    u32_value: u32,
+};
+const arbitrary_output_name: []const u8 = "./output/arbitrary.bin";
+const boxed_type_one_output_name: []const u8 = "./output/boxed_type_one.bin";
+
+pub fn WriteArbitraryType(mem_alloc: std.mem.Allocator) !ArbitraryType {
+    var at: ArbitraryType = .{
+        .size = 25,
+        .line = try mem_alloc.alloc(u8, 25)
+    };
+    // Zero out the u8 array to ensure no trash values
+    // are written to the file.
+    @memset(at.line, 0);
+    const hello_world = "Hello, World!";
+    var ii: u64 = 0;
+    while (ii < hello_world.len) : (ii += 1) {
+        at.line[ii] = hello_world[ii];
+    }
+    const arb_file = try cwd.createFile(arbitrary_output_name, .{.lock = .exclusive});
+    defer arb_file.close();
+    try arb_file.writer().writeAll(std.mem.asBytes(&at.size));
+    try arb_file.writer().writeAll(at.line);
+    return at;
+}
+pub fn ReadArbitraryType(mem_alloc: std.mem.Allocator) !ArbitraryType {
+    const arb_file = try cwd.openFile(arbitrary_output_name,
+        .{.mode = .read_only, .lock = .exclusive});
+    defer arb_file.close();
+    const last_pos = try arb_file.getEndPos();
+    const arbitrary_buffer: []u8 = try mem_alloc.alloc(u8, last_pos);
+    const read_sze = try arb_file.readAll(arbitrary_buffer);
+    if (read_sze == last_pos) {
+        const size = std.mem.bytesToValue(u64, arbitrary_buffer[0..8]);
+        const str: []u8 = try mem_alloc.alloc(u8, size);
+        @memset(str, 0);
+        var ii: u64 = 0;
+        while (ii < size and (ii+8) < last_pos) : (ii += 1) {
+            str[ii] = arbitrary_buffer[ii + 8];
+        }
+        return ArbitraryType{
```

Comment:
In Zig it's idiomatic to just do
```zig
return .{
   .a = X,
   ...
};
```
when the result type is known, meaning that the compiler can infer which struct you're talking about. This can improve the code output using certain optimisations. Also, there's a chance that `X{}` as a syntax may be removed at some point, see https://github.com/ziglang/zig/issues/5038.